docs/asb/diagrams/cli-asb-components-asb-pub-nodes.puml

```diff
@@ -0,0 +1,57 @@
+@startuml
+
+package "User Host" {
+    component [Monero Wallet RPC\n(auto installed)] as CliMonRpc
+    () RPC as MonRpcCli
+    component "swap" {
+        component [CLI] as CLI
+        component [Bitcoin Wallet] as CliBitWal
+        component [Monero Wallet] as CliMonWal
```

Comment:
The "Monero Wallet" of the CLI and also the ASB is not conceptually the same as the Bitcoin wallet:

For Bitcoin, the ASB literally controls the keys whereas for Monero, we keys are controlled by `monero-wallet-rpc`.

I think I would just remove that component and link to the RPC interface directly from the ASB component.

Reply:
I agree that the concept of Bitcoin wallet and Monero wallet are different and should not be depicted in the same way. I will remove the Monero wallet from the depiction for now. 

---

Depicting the RPC without a wallet is not completely correct neither.
The ASB does control the Monero wallet because it actually creates the wallet through the RPC. That's why I depicted it like this. If we decide to implement this https://github.com/comit-network/xmr-btc-swap/issues/382 the Monero wallet is only accessible through the ASB (unless we allow printing the password).
Both the Bitcoin and the Monero wallet are managed by the ASB, but on different abstraction layers.